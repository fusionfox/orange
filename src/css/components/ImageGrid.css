.ImageGrid {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  margin: 30px;
}

.ImageGrid__Thumbnail {
  margin-right: 10px;
  margin-bottom: 10px;
}
