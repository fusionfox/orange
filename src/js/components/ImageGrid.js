import React from 'react'
import '../../css/components/ImageGrid.css';
import classNames from 'classnames';
import Thumbnail from './Thumbnail'

const ImageGrid = (props) => {
  let thumbnails = props.images.map((image) => {
    return <Thumbnail key={image.id} className='ImageGrid__Thumbnail' href={image.href} alt={image.alt} />
  })

  return (
    <div className={classNames(props.className, 'ImageGrid')}>
      {thumbnails}
    </div>
  )
}

export default ImageGrid
