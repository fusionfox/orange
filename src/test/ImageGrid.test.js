import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import ImageGrid from '../js/components/ImageGrid'

test('ImageGrid rendered correctly', () => {
  const images = [{
    id: 'PIA00231',
    title: 'Photo of the moon',
    description: 'This is a photo that we took of the moon',
    thumbnail: `https://images-assets.nasa.gov/image/PIA00231/PIA00231~thumb.jpg`,
    large: `https://images-assets.nasa.gov/image/PIA00231/PIA00231~large.jpg`
  },
  {
    id: 'PIA03149',
    title: 'Photo of the sun',
    description: 'This is a photo that we took of the sun',
    thumbnail: `https://images-assets.nasa.gov/image/PIA03149/PIA03149~thumb.jpg`,
    large: `https://images-assets.nasa.gov/image/PIA03149/PIA03149~large.jpg`
  }]

  const component = renderer.create(
    <MemoryRouter>
      <ImageGrid images={images} />
    </MemoryRouter>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});
