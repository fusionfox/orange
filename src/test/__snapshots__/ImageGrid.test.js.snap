// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageGrid rendered correctly 1`] = `
<ul
  className="ImageGrid"
>
  <li
    className="ImageGrid__Thumbnail Thumbnail"
  >
    <a
      href="/asset/PIA00231"
      onClick={[Function]}
    >
      <img
        alt="Photo of the moon"
        className="Thumbnail__image"
        src="https://images-assets.nasa.gov/image/PIA00231/PIA00231~thumb.jpg"
      />
    </a>
  </li>
  <li
    className="ImageGrid__Thumbnail Thumbnail"
  >
    <a
      href="/asset/PIA03149"
      onClick={[Function]}
    >
      <img
        alt="Photo of the sun"
        className="Thumbnail__image"
        src="https://images-assets.nasa.gov/image/PIA03149/PIA03149~thumb.jpg"
      />
    </a>
  </li>
</ul>
`;
